src/functions.rs

```diff
@@ -159,36 +138,33 @@ pub fn segment_intersection(
     bx1: f64,
     by1: f64,
 ) -> GResult<Option<(f64, f64)>> {
-    match ContextHandle::init() {
-        Ok(context) => unsafe {
-            let mut cx = 0.;
-            let mut cy = 0.;
-
-            let ret = GEOSSegmentIntersection_r(
-                context.as_raw(),
-                ax0,
-                ay0,
-                ax1,
-                ay1,
-                bx0,
-                by0,
-                bx1,
-                by1,
-                &mut cx,
-                &mut cy,
-            );
-            if ret == -1 {
-                Ok(None)
-            } else if ret == 0 {
-                Ok(Some((cx, cy)))
-            } else {
-                Err(Error::GenericError(
-                    "GEOSSegmentIntersection_r failed".to_owned(),
-                ))
-            }
-        },
-        Err(e) => Err(e),
-    }
+    with_context(|ctx| unsafe {
+        let mut cx = 0.;
+        let mut cy = 0.;
+
+        let ret = GEOSSegmentIntersection_r(
+            ctx.as_raw(),
```

Comment:
Is it needed here? Couldn't `with_context` be called inside `GEOSSegmentIntersection_r` methods directly?

Reply:
`GEOSSegmentIntersection_r` is a GEOS C API, how would that work?

Reply:
Ah it's a function. My bad, then that answers my question.